hamilton/plugins/jupyter_magic.py

```diff
@@ -163,95 +194,160 @@ def determine_notebook_type() -> str:
 
 @magics_class
 class HamiltonMagics(Magics):
-    """Magics to facilitate Hamilton development in Jupyter notebooks"""
+    """Magics to facilitate interactive Hamilton development in notebooks."""
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.driver = None
+        self.driver_name = None
+        self.builder = None
+        self.external_modules = list()
+        self.final_vars = list()  # empty list will be converted to "all variables"
+
+        if not hasattr(self, "notebook_env"):
+            self.notebook_env = determine_notebook_type()
 
     @magic_arguments()  # needed on top to enable parsing
+    @argument("module_name", help="Name for the module defined in this cell.")
     @argument(
-        "-m", "--module_name", help="Module name to provide. Default is jupyter_module."
-    )  # keyword / optional arg
+        "-d",
+        "--display",
+        nargs="?",
+        const=True,
+        help="Display the dataflow. The argument is the variable name of a dictionary of visualization kwargs; else {}.",
+    )
     @argument(
-        "-c", "--config", help="JSON config, or variable name containing config to use."
-    )  # keyword / optional arg
+        "-x",
+        "--execute",
+        nargs="?",
+        const=True,
+        help="Execute the dataflow. The argument is the variable name of a list of nodes; else execute all nodes.",
+    )
     @argument(
-        "-r", "--rebuild-drivers", action="store_true", help="Flag to rebuild drivers"
-    )  # Flag / optional arg
+        "-b",
+        "--builder",
+        help="Builder to which the module will be added and used for execution. Allows to pass Config and Adapters",
+    )
     @argument(
-        "-d", "--display", action="store_true", help="Flag to visualize dataflow."
-    )  # Flag / optional arg
+        "-i",
+        "--inputs",
+        help="Execution inputs. The argument is the variable name of a dict of inputs; else {}.",
+    )
     @argument(
-        "-v", "--verbosity", type=int, default=1, help="0 to hide. 1 is normal, default"
-    )  # keyword / optional arg
+        "-o",
+        "--overrides",
+        help="Execution overrides. The argument is the variable name of a dict of overrides; else {}.",
+    )
+    @argument(
+        "-h",
```

Comment:
I think most people will think `-h` is help?

Reply:
I'll remove the short form `-h` and keep `--hide_result`. I wanted a different word family from "display, view, visualize" since we use those for viewing the DAG in other contexts.